### DISCUSSION TITLE
How to use Hardened JS with react/vue/etc?

### DISCUSSION THREAD

**Discussion by dckc** [Agoric/agoric-sdk#6494] [Q&A]
front end bundlers tend to mangle the ses shim.

@turadg @samsiegart @mhofman @kriskowal how do we get around this?

I gather the state of the art is dapp-psm:

https://github.com/Agoric/dapp-psm/blob/e087c78f8057204d8b325f740efe07e54266d82f/vite.config.ts#L5-L8

but I tried to reproduce that in a project of my own without luck. Here's hoping I find time to re-trace my steps to share the symptoms in more detail.

**Comment by tgrecojs** [comment]
Did you ever get to the bottom of this? @dckc The project we're building has a vite.config.js file which mirrors the one you linked to above, and there haven't been any issues with it.

Integrating hardened js into the context of a next.js application is one scenario I've had trouble with over the last few years, so I put together a barebones repository that I could reference if I ever needed a starting point. - 


[github.com/tgrecojs/hardened-nextjs](https://github.com/tgrecojs/hardened-nextjs)


### `Object.isFrozen` 
Executing `npm install && npm run start`  within the repo above will get you to this starting point.


https://user-images.githubusercontent.com/6646552/216515218-6b8d9c7b-5543-4eab-94d8-75e7a39cf0ae.mp4

**Comment by dckc** [comment]
[installSesLockdown.ts](https://github.com/samsiegart/simple-game/blob/main/src/installSesLockdown.ts) seems to be state-of-the-art.

Here's hoping one of us finds a moment to explain how it works.

**Comment by mhofman** [comment]
[Vetted shims support in ses-shim](https://github.com/endojs/endo/issues/422) is [making progress](https://github.com/endojs/endo/pull/1705). That should make it easier to support these environments.
